### DISCUSSION TITLE
Wrong format rabbitmqctl list_permissions

### DISCUSSION THREAD

**Discussion by letanloc1998** [rabbitmq/rabbitmq-server#2948] [Other] [answered]
```shell
rabbitmqctl list_permissions -p vhost-gateway-desktop-sendktop-send
Listing permissions for vhost "vhost-gateway-desktop-send" ...
user    configure       write   read
user-gateway-desktop    .*      .*      ^$
user-service    .*      ^$      .*
```

wrong tab header when run command `rabbitmqctl list_permissions`

Thank

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment, marked as answer]
I'm afraid I do not see what exactly is "wrong" here, perhaps you should elaborate.

The headers are not padded (aligned) but they do match the actual columns. This commands supports two formatters that make it easier to parse or read the output:

## Padded Table with Borders

``` shell
rabbitmqctl list_permissions --formatter=pretty_table -q
# => ┌────────┬───────────┬───────┬──────┐
# => │ user   │ configure │ write │ read │
# => ├────────┼───────────┼───────┼──────┤
# => │ guest2 │ .*        │ .*    │ .*   │
# => ├────────┼───────────┼───────┼──────┤
# => │ guest  │ .*        │ .*    │ .*   │
# => └────────┴───────────┴───────┴──────┘
```

## JSON

``` shell
rabbitmqctl list_permissions --formatter=json | jq
```

that outputs

``` json
[
  {
    "user": "guest2",
    "configure": ".*",
    "write": ".*",
    "read": ".*"
  },
  {
    "user": "guest",
    "configure": ".*",
    "write": ".*",
    "read": ".*"
  }
]
```
